src/components/forms/application/v2/bank-account-connection.tsx

```diff
@@ -20,16 +363,374 @@ const BankAccountConnection = () => {
+                  </div>
+                ))}
+
+                {/* Add more files button */}
+                <label htmlFor="add-more-files" className="block">
+                  <input
+                    id="add-more-files"
+                    type="file"
+                    className="hidden"
+                    accept=".pdf,.png,.doc,.docx"
+                    multiple
+                    onChange={handleFileSelect}
+                  />
+                  <div className="border-2 border-dashed border-gray-300 rounded-lg p-4 text-center hover:border-gray-400 transition-colors cursor-pointer">
+                    <p className="text-sm text-gray-600">
+                      + Add more files (you have {uploadedFiles.length}/6)
+                    </p>
+                  </div>
+                </label>
+              </div>
+            )}
+
+            {/* Error message */}
+            {uploadError && (
+              <div className="bg-red-50 border border-red-200 rounded-lg p-4">
+                <p className="text-sm text-red-600">{uploadError}</p>
+              </div>
+            )}
+
+            {/* Submit button */}
+            <button
+              onClick={handleManualUploadSubmit}
+              disabled={uploadedFiles.length < 6 || isUploading}
+              className="w-full bg-gray-800 text-white py-3 px-6 rounded-lg font-semibold hover:bg-gray-700 transition-colors disabled:opacity-50 disabled:cursor-not-allowed"
+            >
+              {isUploading ? 'Uploading...' : 'Submit documents'}
+            </button>
+          </div>
+          <div className="text-center mt-4">
+            <button
+              onClick={() => handleMethodSelect('')}
+              className="text-blue-600 hover:text-blue-800 underline text-sm"
+            >
+              Connect to bank instead
+            </button>
+          </div>
+        </div>
+      );
+    }
+
+    if (localFormData.bankConnectionMethod === 'flinks') {
+      return (
+        <div>
+          <h2 className="text-2xl font-bold text-gray-800 mb-4">
+            Connect Your Bank Account
+          </h2>
+
+          <div className="mb-8" style={{ maxHeight: '600px' }}>
+            {isLoading && (
+              <div
+                className="flex justify-center items-center flex-col gap-4 no-scrollbar "
+                style={{ minHeight: '600px' }}
+              >
+                <p className="text-sm text-gray-500">Connecting to Flinks...</p>
+                <div className="animate-spin rounded-full h-12 w-12 border-t-2 border-b-2 border-gray-900"></div>
+              </div>
+            )}
+            <iframe
+              src={`${process.env.NEXT_PUBLIC_FLINKS_IFRAME_URL}/v2?customerName=Keep&daysOfTransactions=Days365&scheduleRefresh=false&consentEnable=true&detailsAndStatementEnable=true&monthsOfStatements=Months12&enhancedMFA=false&maximumRetry=3&tag=${formattedFlinksTags}${skipFlinks ? '&demo=true' : ''}`}
```

Comment:
Any chance we can improve reading on this by doing something like:



```suggestion
...
 const flinksConnectionParams = new URLSearchParams({
      customerName: 'Keep',
      daysOfTransactions: 'Days365',
    ...
    });
...

              src={`${process.env.NEXT_PUBLIC_FLINKS_IFRAME_URL}/v2${flinksConnectionParams.toString()}`}
```